### ISSUE TITLE
When I use Acrobat, I get the same error when printing a PDF. The only requirement is that there is a "path" in the code.

### ISSUE THREAD

**Issue by Lucas-C** [py-pdf/fpdf2#1144]
When I use Acrobat, I get the same error when printing a PDF. The only requirement is that there is a "path" in the code.

Minimal test code:

```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()

with pdf.new_path() as path:
    path.move_to(1, 1)
    path.line_to(9, 9)
    path.close()

pdf.output("test.pdf")
```
Then print test.pdf using Acrobat reader. The error should appear right after printing.
[test.pdf](https://github.com/py-pdf/fpdf2/files/14918697/test.pdf)

The problem persists when pdf.compress = False

_Originally posted by @GerardoAllende in https://github.com/py-pdf/fpdf2/issues/960#issuecomment-2045092620_

**Comment by Lucas-C** [comment]
Hi @GerardoAllende

I was able to open the `test.pdf` file produced by this script in both [Sumatra PDF reader](https://www.sumatrapdfreader.org/) and Adobe Acrobat reader 2024.001.20643 under Windows.

Sorry, I cannot reproduce your problem! 😢

**Comment by GerardoAllende** [comment]
The problem shows only when printing. Only with Acrobat. Screenshot attached. Tested in Windows 7 and 10.
![1](https://github.com/py-pdf/fpdf2/assets/8699267/1961b70c-c0b8-47e7-9596-853eb2968584)
![2](https://github.com/py-pdf/fpdf2/assets/8699267/7a9dbde4-bb4d-47b1-b715-c0bf6d683983)

**Comment by GerardoAllende** [comment]
The PDFs in https://py-pdf.github.io/fpdf2/Drawing.html for example:

https://py-pdf.github.io/fpdf2/drawing/demo-1.pdf

Are not showing the nag message box in Acrobat when printing.

But, if I create the PDF using the sample code in the page, the PDF shows the message when printing. The PDFs created and the ones in the page are different. Do you know if the PDFs in the page were post processed? Or do you remember how they were created? I'm using release 2.7.8.

**Comment by GerardoAllende** [comment]
Manually edited the PDF, changed this:

```
stream
2 J
0.57 w
q 2.8346 0 0 -2.8346 0 28.3465 cm /GS0 gs [] 0 d q 2 2 m 8 8 l 2 8 l 8 2 l h B Q Q
endstream
```

Deleted  /GS0 gs [] 0 d
Result:
```
stream
2 J
0.57 w
q 2.8346 0 0 -2.8346 0 28.3465 cm                q 2 2 m 8 8 l 2 8 l 8 2 l h B Q Q
endstream
```

And the PDF is not showing the nag message box when printing.

Anybody knows what means the deleted part? /GS0 gs [] 0 d

**Comment by GerardoAllende** [comment]
Well, temporary solution found:

Comment lines 1439 to 1435 of drawing.py

**Comment by andersonhc** [comment]
@GerardoAllende would you be interested in submitting a PR?

**Comment by GerardoAllende** [comment]
Yes, thanks! May we reduce the if to:

```
if self.stroke_dash_pattern and (
   self.stroke_dash_pattern is not self.INHERIT
        ):
```
?

**Comment by andersonhc** [comment]
> Yes, thanks! May we reduce the if to:
> 
> ```
> if self.stroke_dash_pattern and (
>    self.stroke_dash_pattern is not self.INHERIT
>         ):
> ```
> 
> ?

Definitely, yes.

**Comment by gmischler** [comment]
Fixed in #1145
